### DISCUSSION TITLE
Issue with Identifying Individual Databases for Multiple Applications

### DISCUSSION THREAD

**Discussion by vishalkadam18** [apache/skywalking#12521] [Q&A]
Dear Team,

While monitoring two distinct applications in Apache SkyWalking, both having their own databases but sharing the same database name in their application.properties files, the visual database section displays only one database name. This makes it difficult to determine which application’s database is being visualized. We need clarification on how to distinguish between these databases in the UI and whether SkyWalking provides any mapping or configuration options to identify the database names corresponding to individual services, ensuring accurate monitoring and analysis.

For example, we have two applications, application-1 and application-2, both configured with the same database URL in their application.properties files, such as spring.datasource.url=jdbc:mysql://localhost:3306/springboot. In the visual database section of Apache SkyWalking, only one entry for localhost:3306 appears. This makes it difficult to determine which application's database is being monitored and how to monitor both databases individually.

Thanks
Vishal

**Comment by wu-sheng** [comment]
I think we treat these as the same thing because you use the same name. 
In current cloud name env, the service name should be unique.

**Comment by vishalkadam18** [comment]
Thank you for the clarification. I understand that service names should be unique, and in our setup, each service does indeed have a unique name. However, the issue we’re facing is related to the database URLs, which might be the same across different services.

In our case, we have two distinct applications (application-1 and application-2), both configured with the same database URL (jdbc:mysql://localhost:3306/springboot). This results in the visual database section of Apache SkyWalking showing only one entry for this URL. This makes it challenging to determine which service is interacting with which specific database, especially when the URLs are identical.

To address this, it would be helpful to understand if there is any way to differentiate or map the database interactions in SkyWalking based on the service name, despite the shared database URL. Are there any configuration options or settings in SkyWalking that can help in distinguishing the database interactions from different services, perhaps through custom tags or additional metadata?

Thanks
Vishal.

**Reply by wu-sheng** [reply]
That is by design not to distinct because it should be the same service. Otherwise, the topology will show two nodes which would be a bug.
